src/main/java/com/zendesk/maxwell/producer/MaxwellKafkaProducer.java

```diff
@@ -177,6 +181,7 @@ public MaxwellKafkaProducerWorker(MaxwellContext context, Properties kafkaProper
 		this.ddlPartitioner = makeDDLPartitioner(hash, partitionKey);
 		this.ddlTopic =  context.getConfig().ddlKafkaTopic;
 		this.deadLetterTopic = context.getConfig().deadLetterTopic;
+		this.deadLetterQueue = this.deadLetterTopic == null ? null : new ConcurrentLinkedQueue<>();
```

Comment:
seems overly optimized, just init the thing.

Reply:
I don't care about the memory, this is just so we can skip `poll()`ing it if it's `null`. Otherwise we'll `poll()` an unused queue on every message when the feature is turned off. But if you're not worried, I'm good either way 🤷‍♂️